features/sharding/distsql/handler/src/main/java/org/apache/shardingsphere/sharding/distsql/handler/update/AlterShardingTableReferenceRuleExecutor.java

```diff
@@ -91,7 +92,8 @@ private Collection<String> getToBeAlteredRuleNames(final ShardingRuleConfigurati
     
     private void checkToBeReferencedShardingTablesExisted(final AlterShardingTableReferenceRuleStatement sqlStatement) {
         Collection<String> existedShardingTables = getCurrentShardingTables();
-        Collection<String> notExistedShardingTables = sqlStatement.getTableNames().stream().filter(each -> !containsIgnoreCase(existedShardingTables, each)).collect(Collectors.toSet());
+        Collection<String> notExistedShardingTables =
+                sqlStatement.getTableNames().stream().filter(each -> !CollectionUtils.containsIgnoreCase(existedShardingTables, each)).collect(Collectors.toSet());
```

Comment:
Can you consider CaseInsensitiveSet or CaseInsensitiveMap?

Reply:
@strongduanmu Thanks for the tip, I'll try it

Reply:
Done #30180